### PR TITLE
Support expanded and flat objects

### PR DESCRIPTION
**Pull request by mtojek**
Issue: https://github.com/elastic/package-spec/issues/55

For ease of merging, I branched out from this PR: https://github.com/elastic/package-spec/pull/54 (if that's an issue I can branch from master).

### CODE REVIEW

**Comment by mtojek** [comment]
This is another place where `mapstr.go` is needed. I think we need to put it in a good place.

**Comment by ycombinator** [comment]
> For ease of merging, I branched out from this PR: #54 (if that's an issue I can branch from master).

It depends on how much longer we're going to wait on #54, which depends on how much longer we want to wait for the associated ES issue to at least be discussed. 

I would prefer to branch this PR (#56) from `master` so there's no dependency on #54.

**Comment by mtojek** [comment]
I was thinking the other way round. Let's consider #54 as workaround and once the discussion with ES is finalized we'll adjust the implementation, but I don't mind splitting these PRs.

**Comment by ycombinator** [comment]
That plan also works for me. In that case don’t bother splitting out this PR. I will review #54 now so we can get it in shortly. Then you can just rebase this PR on `master`.

**Comment by mtojek** [comment]
Ok, I rebased against master. It's ready for review.

File: code/go/internal/validator/test/packages/missing_pipeline_dashes/manifest.yml [hunk 1]
```diff
@@ -5,7 +5,8 @@ description: This package is good.
 version: 0.1.2
 release: beta
 conditions:
-  kibana.version: '^7.9.0'
+  kibana:
+    version: '^7.9.0'
```
Comment (ycombinator) [inline code/go/internal/validator/test/packages/missing_pipeline_dashes/manifest.yml:9]: I assume the purpose of the new `expanded` test package is to test the case where the kibana version condition is specified like this:

```
kibana:
  version: ...
```

Given that, should we leave this line as-is (flattened syntax) to make sure both syntaxes are being tested?

Comment (mtojek) [reply code/go/internal/validator/test/packages/missing_pipeline_dashes/manifest.yml]: You're right. Fixed.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Pull request merged**